### PR TITLE
Do not render unstable items for rustc doc

### LINKED ISSUE #54682: Docs: Hide nightly-only messages on doc.rust-lang.org/nightly-rustc
When reading the nightly-rustc docs, all the yellow boxes saying they're experimental is entirely noise. It'd be nice to remove them.

### PR DESCRIPTION
**Pull request by tesuji**
See the zulip conversation at https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/rustc.20doc.3A.20.22internal.20compiler.20API.22.20warns.20are.20everywhere!/near/203850782

Before: 
image
After:  
image

Nothing changes in unstable items of std:
Before: 
image
After: 
image

Closes  #54682

### CODE REVIEW

File: src/librustdoc/externalfiles.rs [hunk 1]
```diff
@@ -37,14 +37,14 @@ impl ExternalHtml {
         let bc = format!(
             "{}{}",
             bc,
-            Markdown(&m_bc, &[], id_map, codes, edition, playground).to_string()
+            Markdown(&m_bc, &[], id_map, codes, edition, playground).into_string()
```
Comment (ollie27) [inline src/librustdoc/externalfiles.rs:40]: Can this change be done in a separate PR? It seems unrelated to this one.

File: src/test/rustdoc/internal.rs [hunk 2]
```diff
@@ -1,12 +0,0 @@
-// compile-flags: -Z force-unstable-if-unmarked
-
-// @matches internal/index.html '//*[@class="docblock-short"]/span[@class="stab internal"]' \
-//      'Internal'
```
Comment (ollie27) [inline src/test/rustdoc/internal.rs:4]: This test should be modified to check that the unstable marker is not added for `rustc_private`.

**Review by ollie27 — COMMENTED** [review]
This PR will fix  #54682.

The now unused CSS should be removed:

https://github.com/rust-lang/rust/blob/c714eae0e3b4ba263c193d54f6e46bb9a1feb2cc/src/librustdoc/html/static/themes/light.css#L176-L178 and https://github.com/rust-lang/rust/blob/c714eae0e3b4ba263c193d54f6e46bb9a1feb2cc/src/librustdoc/html/static/themes/light.css#L218 for every theme.

File: src/librustdoc/externalfiles.rs (hunk 1 above)
Comment (lzutao) [reply src/librustdoc/externalfiles.rs:40]: Split into #74359. Will update this PR soon.

**Comment by GuillaumeGomez** [comment]
Also: can you remove the changes coming from #74359 please? It makes the changes more noisy than necessary.

File: src/test/rustdoc/internal.rs (hunk 2 above)
Comment (lzutao) [reply src/test/rustdoc/internal.rs:4]: Done

**Comment by lzutao** [comment]
Resolved comment reviews.

File: src/librustdoc/html/render.rs [hunk 3]
```diff
@@ -2285,15 +2288,13 @@ fn short_stability(item: &clean::Item, cx: &Context) -> Vec<String> {
...
-        let mut message = if is_rustc_private {
-            "<span class='emoji'>⚙️</span> This is an internal compiler API."
-        } else {
-            "<span class='emoji'>🔬</span> This is a nightly-only experimental API."
-        }
-        .to_owned();
+    // Render unstable items. But don't render "rustc_private" crates (internal compiler crates).
+    // Those crates are permanently unstable so it makes no sense to render "unstable" everywhere.
+    if let Some(stab) = item.stability.as_ref().filter(|stab| {
+        stab.level == stability::Unstable && stab.feature.as_deref() != Some("rustc_private")
```
Comment (Mark-Simulacrum) [inline src/librustdoc/html/render.rs:2294]: nit: we should be comparing against `rustc_span::sym::rustc_private` to avoid hitting the string interner lock here.

**Comment by Mark-Simulacrum** [comment]
r=me with nit resolved (where I left the comment and elsewhere in modified code)

File: src/librustdoc/html/render.rs (hunk 3 above)
Comment (lzutao) [reply src/librustdoc/html/render.rs:2294]: I would love to. But [`stab.feature`](https://doc.rust-lang.org/nightly/nightly-rustc/rustdoc/clean/types/struct.Stability.html) is `Option<String>`. I'm planning for merging in https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/Merge.20Stability.20in.20rustc_attr.20and.20rustdoc .

Comment (Mark-Simulacrum) [reply src/librustdoc/html/render.rs:2294]: Oh right this is rustdoc-cleaned stability. Fine as is then.
